src/Catty/FormulaEditor/ViewController/FormulaEditorViewController+Section.swift

```diff
@@ -22,49 +22,16 @@
 
 extension FormulaEditorViewController {
 
-    @objc func initMathSection(scrollView: UIScrollView, buttonHeight: CGFloat) -> [UIButton] {
-        let items = formulaManager.formulaEditorItemsForMathSection(spriteObject: object)
-        return initWithItems(formulaEditorItems: items, scrollView: scrollView, buttonHeight: buttonHeight, fullButtonWidth: true)
-    }
-
-    @objc func initLogicSection(scrollView: UIScrollView, buttonHeight: CGFloat) -> [UIButton] {
-        let items = formulaManager.formulaEditorItemsForLogicSection(spriteObject: object)
-        return initWithItems(formulaEditorItems: items, scrollView: scrollView, buttonHeight: buttonHeight, fullButtonWidth: false)
-    }
-
-    @objc func initObjectSection(scrollView: UIScrollView, buttonHeight: CGFloat) -> [UIButton] {
-        let items = formulaManager.formulaEditorItemsForObjectSection(spriteObject: object)
-        return initWithItems(formulaEditorItems: items, scrollView: scrollView, buttonHeight: buttonHeight, fullButtonWidth: true)
-    }
-
-    @objc func initSensorSection(scrollView: UIScrollView, buttonHeight: CGFloat) -> [UIButton] {
-        let items = formulaManager.formulaEditorItemsForDeviceSection(spriteObject: object)
-        return initWithItems(formulaEditorItems: items, scrollView: scrollView, buttonHeight: buttonHeight, fullButtonWidth: true)
-    }
-
-    private func initWithItems(formulaEditorItems: [FormulaEditorItem], scrollView: UIScrollView, buttonHeight: CGFloat, fullButtonWidth: Bool) -> [UIButton] {
-        var button: UIButton?
-        var buttons = [UIButton]()
-
-        for item in formulaEditorItems {
-            button = buttonForScrollView(item: item, scrollView: scrollView, previousButton: button, buttonHeight: buttonHeight, fullButtonWidth: fullButtonWidth)
-            buttons.append(button!)
+    func formulaEditorItemSelected(item: FormulaEditorItem) {
```

Comment:
How about integrating these decisions into `FormulaEditorSectionViewController` directly? Now that the view controller is very small and since that is still part of the section, we could integrate it there directly. The public interface to the `FormulaEditor` would then be `handleInput` which would be very readable and clean. And very good to be tested using unit test and test doubles (please add a unit test for that).

Reply:
Hi Michael, I have pushed a new commit to this PR with all the changes you suggested, including this one. However, will you please give some suggestions on how exactly should I write unit tests for this? Also, whether I have understood and implemented this properly.

Reply:
I have added an example test case for `FormulaEditorSectionViewControllerTests`. If you agree with this test, could you please add it for the other sections as well?

Reply:
Thank you very much! I have pushed a new commit with tests and multiple other changes. Please review it.